Mathlib/AlgebraicGeometry/Morphisms/UnderlyingMap.lean

```diff
@@ -0,0 +1,147 @@
+/-
+Copyright (c) 2022 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.Topology.LocalAtTarget
+import Mathlib.AlgebraicGeometry.Morphisms.Constructors
+
+/-!
+
+## Properties on the underlying functions of morphisms of schemes.
+
+This file includes various results on properties of morphisms of schemes that come from properties
+of the underlying map, including
+
+- `Injective`
+- `Surjective`
+- `IsOpenMap`
+- `IsClosedMap`
+- `Embedding`
+- `OpenEmbedding`
+- `ClosedEmbedding`
+
+
+-/
+
+open CategoryTheory
+
+namespace AlgebraicGeometry
+
+universe u
+
+section Injective
+
+variable {X Y Z : Scheme.{u}} (f : X ⟶ Y) (g : Y ⟶ Z)
+
+instance : MorphismProperty.RespectsIso (topologically Function.Injective) :=
+  topologically_respectsIso _ (fun e ↦ e.injective) (fun _ _ hf hg ↦ hg.comp hf)
+
+instance injective_isLocalAtTarget : IsLocalAtTarget (topologically Function.Injective) := by
+  refine topologically_isLocalAtTarget _ (fun _ s h ↦ h.restrictPreimage s)
+    fun f ι U H _ hf x₁ x₂ e ↦ ?_
+  obtain ⟨i, hxi⟩ : ∃ i, f x₁ ∈ U i := by simpa using congr(f x₁ ∈ $H)
+  exact congr(($(@hf i ⟨x₁, hxi⟩ ⟨x₂, show f x₂ ∈ U i from e ▸ hxi⟩ (Subtype.ext e))).1)
+
+end Injective
+
+section Surjective
+
+variable {X Y Z : Scheme.{u}} (f : X ⟶ Y) (g : Y ⟶ Z)
+
+/-- A morphism of schemes is surjective if the underlying map is. -/
+@[mk_iff]
+class Surjective : Prop where
+  surj : Function.Surjective f.1.base
+
+lemma surjective_eq_topologically :
+    @Surjective = topologically Function.Surjective := by ext; exact surjective_iff _
+
+lemma Scheme.Hom.surjective (f : X.Hom Y) [Surjective f] : Function.Surjective f.1.base :=
+  Surjective.surj
+
+instance (priority := 100) [IsIso f] : Surjective f := ⟨f.homeomorph.surjective⟩
+
+instance [Surjective f] [Surjective g] : Surjective (f ≫ g) := ⟨g.surjective.comp f.surjective⟩
+
+lemma Surjective.of_comp [Surjective (f ≫ g)] : Surjective g where
+  surj := Function.Surjective.of_comp (g := f.1.base) (f ≫ g).surjective
+
+lemma Surjective.comp_iff [Surjective f] : Surjective (f ≫ g) ↔ Surjective g :=
+  ⟨fun _ ↦ of_comp f g, fun _ ↦ inferInstance⟩
+
+instance : MorphismProperty.RespectsIso @Surjective :=
+  surjective_eq_topologically ▸ topologically_respectsIso _ (fun e ↦ e.surjective)
+    (fun _ _ hf hg ↦ hg.comp hf)
+
+instance surjective_isLocalAtTarget : IsLocalAtTarget @Surjective := by
+  have := AlgebraicGeometry.instRespectsIsoSchemeSurjective
```

Comment:
```suggestion
  have : MorphismProperty.RespectsIso @Surjective := inferInstance
```
I don't know how stable it is to rely on the automatically generated instance name.

Reply:
(alternatively: give the `RespectsIso` instance a name)